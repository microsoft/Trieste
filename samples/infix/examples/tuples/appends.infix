print "empty and empty" append((,), (,));
print "empty append" append(,);
print "unary append" append((,),);
print "nested appends" append((1, 2), append((3,), (4,)));