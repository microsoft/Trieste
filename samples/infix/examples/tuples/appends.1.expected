//! --enable-tuples --tuples-require-parens parse_only
//! --enable-tuples --tuples-require-parens --use-parser-tuples parse_only
//! --enable-tuples parse_only
(top
  {}
  (infix-calculation
    {}
    (infix-output
      (infix-string 17:"empty and empty")
      (infix-expression
        (infix-append
          (infix-expression
            (infix-tuple))
          (infix-expression
            (infix-tuple)))))
    (infix-output
      (infix-string 14:"empty append")
      (infix-expression
        (infix-append)))
    (infix-output
      (infix-string 14:"unary append")
      (infix-expression
        (infix-append
          (infix-expression
            (infix-tuple)))))
    (infix-output
      (infix-string 16:"nested appends")
      (infix-expression
        (infix-append
          (infix-expression
            (infix-tuple
              (infix-expression
                (infix-int 1:1))
              (infix-expression
                (infix-int 1:2))))
          (infix-expression
            (infix-append
              (infix-expression
                (infix-tuple
                  (infix-expression
                    (infix-int 1:3))))
              (infix-expression
                (infix-tuple
                  (infix-expression
                    (infix-int 1:4)))))))))))
